tornado/test/httpserver_test.py

```diff
@@ -441,6 +441,18 @@ def test_chunked_request_body(self):
         headers, response = self.wait()
         self.assertEqual(json_decode(response), {u'foo': [u'bar']})
 
+    def test_invalid_content_length(self):
+        with ExpectLog(gen_log, '.*Only integer Content-Length is allowed'):
+            self.stream.write(b"""\
+POST /echo HTTP/1.1
+Content-Length: foo
+
+bar
+
+""".replace(b"\n", b"\r\n"))
+            self.io_loop.add_timeout(datetime.timedelta(seconds=0.01), self.stop)
```

Comment:
Tests that rely on timing like this are flaky. Can you do something like `self.stream.read_until_close(self.stop)` instead? (does the server just close the connection, or does it write a 400 error response?)